contrib/caldav/calcurse-caldav.py

```diff
@@ -421,17 +420,14 @@ def push_object(conn, objhash):
     if not headers:
         return None
 
-    etag = None
-    headerdict = dict(headers)
-    if 'etag' in headerdict:
-        etag = headerdict['etag']
-    while not etag:
-        etagdict = get_etags(conn, [href])
-        if etagdict:
-            etag = next(iter(etagdict.values()))
+    etagdict = get_etags(conn, [href])
+    if not etagdict:
+        return None
```

Comment:
A few questions:

1. Does this introduce a possible race condition when there's another update after pushing the new item but before retrieving the ETag? Should we only use the response to update `href` but not the ETag (unless it is missing)?
2. This also seems to remove the loop. Can you add a comment explaining why it is removed and how this impacts the behavior of the script?

Reply:
Thank you for the review,

yes, this does simplify the logic too much, I'll push a more conservative edit which should again account for those two points.